client/main.lua

```diff
@@ -92,90 +108,151 @@ local function GetDeliveryLocation()
     SetBlipRouteColour(NpcData.DeliveryBlip, 3)
     NpcData.LastDeliver = route
     local inRange = false
+    local shownTextUI = false
     DeliverZone = lib.zones.sphere({
-        name = "busjobdeliver",
+        name = "qb_busjob_bus_deliver",
         coords = vec3(Config.NPCLocations.Locations[route].x,
         Config.NPCLocations.Locations[route].y, Config.NPCLocations.Locations[route].z),
         radius = 5,
         debug = false,
         onEnter = function()
             inRange = true
-            exports["qb-core"]:DrawText(Lang:t('info.busstop_text'), 'rgb(220, 20, 60)')
+            if not shownTextUI then
+                lib.showTextUI(Lang:t('info.busstop_text'))
+                shownTextUI = true
+            end
             CreateThread(function()
                 repeat
                     Wait(0)
                     if IsControlJustPressed(0, 38) then
-                        local veh = GetVehiclePedIsIn(cache.ped, false)
-                        TaskLeaveVehicle(NpcData.Npc, veh, 0)
+                        TaskLeaveVehicle(NpcData.Npc, cache.vehicle, 0)
                         SetEntityAsMissionEntity(NpcData.Npc, false, true)
                         SetEntityAsNoLongerNeeded(NpcData.Npc)
                         local targetCoords = Config.NPCLocations.Locations[NpcData.LastNpc]
                         TaskGoStraightToCoord(NpcData.Npc, targetCoords.x, targetCoords.y, targetCoords.z, 1.0, -1, 0.0, 0.0)
                         QBCore.Functions.Notify(Lang:t('success.dropped_off'), 'success')
-                        if NpcData.DeliveryBlip ~= nil then
+                        if NpcData.DeliveryBlip then
                             RemoveBlip(NpcData.DeliveryBlip)
                         end
-                        local RemovePed = function(pped)
-                            SetTimeout(60000, function()
-                                DeletePed(pped)
-                            end)
-                        end
-                        RemovePed(NpcData.Npc)
+                        removePed(NpcData.Npc)
                         resetNpcTask()
                         nextStop()
                         TriggerEvent('qb-busjob:client:DoBusNpc')
-                        exports["qb-core"]:HideText()
+                        lib.hideTextUI()
+                        shownTextUI = false
                         DeliverZone:remove()
+                        DeliverZone = nil
                         break
                     end
```

Comment:
This could be extracted and turned into a gate function.

Reply:
Only used once so I don't see the point